### ISSUE TITLE
Recovery flashable ZIPs 

### ISSUE THREAD

**Issue by cRaZy-bisCuiT** [microg/android_packages_apps_GmsCore#980]
Heya,

would it be possible to provide ZIPs that are flashable from recovery? Having a ROM that does not include MicroG the GCM/FCM checkin is lost all the time you update the ROM. That kinda breaks Apps like Signal since it needs to be re-installed all the time to force a checkin. In the process you lose your private keys and all the groups you used to be invited in.

**Comment by cRaZy-bisCuiT** [comment]
In addition: Is there a guide how to integrate MircroG myself if I have a premade ROM?

**Comment by chris42** [comment]
There is [nanodroid](https://nanolx.org/nanolx/nanodroid) from @Nanolx to install it via recovery, there is also a documentation on how to use it.

**Comment by cRaZy-bisCuiT** [comment]
Thanks! But is it possible to only get selected MicroG components installed? The package seems to be bundling a lot of bloatware as well.

**Comment by chris42** [comment]
Yes, you can select it during install and as far as I know even create a configuration file: https://gitlab.com/Nanolx/NanoDroid#installation

**Comment by Nanolx** [comment]
@cRaZy-bisCuiT There's a much smaller microG package aswell.

Besides the Full package does technically not contain bloatware, because it /replaces/ (by default) the ROM apps with FOSS alternatives, so there's not more installed than before (in most cases), thus technically no bloat. Not that it matters, though.

As @chris42 pointed out you can create configuration files which allow you to install only what you want. That applies to all packages.

Also there's @ale5000-git unofficial microG installer.

**Comment by jojo313** [comment]
> Besides the Full package does technically not contain bloatware, because it /replaces/ (by default) the ROM apps with FOSS alternatives, so there's not more installed than before (in most cases), thus technically no bloat. Not that it matters, though.

Zelda ringtones aren't bloat?

**Comment by Nanolx** [comment]
Just don't install them. There I am, giving people the power to choose what to install and yet they complain about audio files of less than 9 MB combined as bloat. *sigh*

Am 28. Dezember 2019 21:44:07 MEZ schrieb jojo313 <notifications@github.com>:
>> Besides the Full package does technically not contain bloatware, because it /replaces/ (by default) the ROM apps with FOSS alternatives, so there's not more installed than before (in most cases), thus technically no bloat. Not that it matters, though.
>
>Zelda ringtones aren't bloat?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/microg/android_packages_apps_GmsCore/issues/980#issuecomment-569449773
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by jojo313** [comment]
> 
> Just don't install them. There I am, giving people the power to choose what to install and yet they complain about audio files of less than 9 MB combined as bloat. *sigh* Am 28. Dezember 2019 21:44:07 MEZ schrieb jojo313 <notifications@github.com>:
> > Besides the Full package does technically not contain bloatware, because it /replaces/ (by default) the ROM apps with FOSS alternatives, so there's not more installed than before (in most cases), thus technically no bloat. Not that it matters, though. Zelda ringtones aren't bloat? -- You are receiving this because you were mentioned. Reply to this email directly or view it on GitHub: [#980 (comment)](https://github.com/microg/android_packages_apps_GmsCore/issues/980#issuecomment-569449773)
> -- Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

Not complaining, just saying that the full package contains stuff which is rather unrelated, so bloat in my mind. 

I'm very happy with the standard small microg package. Danke für deine Arbeit <3
